shared/common-adapters/tabs.tsx

```diff
@@ -45,38 +47,64 @@ const TabText = ({selected, text}: {selected: boolean; text: string}) => (
   </Kb.Box2>
 )
 
-const Tabs = <TitleT extends string>(props: Props<TitleT>) => (
-  <Kb.Box2
-    direction="horizontal"
-    style={Styles.collapseStyles([styles.container, props.style])}
-    alignItems="flex-start"
-    fullWidth={true}
-  >
-    {props.tabs.map((tab: Tab<TitleT>) => {
-      const selected = props.selectedTab === tab.title
-      return (
-        <Kb.ClickableBox
-          onClick={() => props.onSelect(tab.title)}
-          key={tab.title}
-          style={props.clickableBoxStyle}
-        >
-          <Kb.Box2 direction="vertical" style={styles.tabContainer} fullWidth={true}>
-            <Kb.Box style={Styles.collapseStyles([styles.tab, selected && styles.selected, props.tabStyle])}>
-              {tab.icon ? (
-                <Kb.Icon type={tab.icon} style={selected ? styles.iconSelected : styles.icon} />
-              ) : (
-                <TabText selected={selected} text={tab.text ?? capitalize(tab.title)} />
-              )}
-              {!!tab.badgeNumber && <Kb.Badge badgeNumber={tab.badgeNumber} badgeStyle={styles.badge} />}
-            </Kb.Box>
-            <Kb.Divider style={selected ? styles.dividerSelected : styles.divider} />
-          </Kb.Box2>
-        </Kb.ClickableBox>
-      )
-    })}
-    {props.showProgressIndicator && <Kb.ProgressIndicator style={styles.progressIndicator} />}
-  </Kb.Box2>
-)
+const Tabs = <TitleT extends string>(props: Props<TitleT>) => {
+  const mobileTabMode = props.mobileTabModeOverride ?? (props.tabs.length > 3 ? 'scroll' : 'distribute')
```

Comment:
:sheep: might clean up some of the checks below if this were undefined on non-mobile